packages/nextjs/src/editing/editing-render-middleware.ts

```diff
@@ -200,124 +142,59 @@ export class EditingRenderMiddleware extends RenderMiddlewareBase {
       });
     }
 
-    if (isDesignLibraryMode(mode)) {
-      res.setPreviewData(
-        {
-          itemId: query.sc_itemid,
-          componentUid: query.sc_uid,
-          renderingId: query.sc_renderingId,
-          language: query.sc_lang,
-          site: query.sc_site,
-          mode,
-          dataSourceId: query.dataSourceId,
-          version: query.sc_version,
-        } as DesignLibraryRenderPreviewData,
-        {
-          maxAge: 3,
-        }
-      );
-    } else {
-      res.setPreviewData(
-        {
-          site: query.sc_site,
-          itemId: query.sc_itemid,
-          language: query.sc_lang,
-          // for sc_variantId we may employ multiple variants (page-layout + component level)
-          variantIds: query.sc_variant?.split(',') || [DEFAULT_VARIANT],
-          version: query.sc_version,
-          mode: query.mode,
-          layoutKind: query.sc_layoutKind,
-        } as EditingPreviewData,
-        // Cache the preview data for 3 seconds to ensure the page is rendered with the correct preview data not the cached one
-        {
-          maxAge: 3,
-        }
-      );
-    }
+    const previewDataParams = getEditingParams(query as { [key: string]: string });
+
+    // With the editing optimization work done, we can cut the dependency on nextjs preview mode
+    // @TODO in next major release: pass preview data via query string, variantIds will be a single string too
```

Comment:
passing query string parameters in Pages router won't work for SSG projects. Next simply doesn't provide them to getStaticProps function.